pkg/broker/ingress/ingress_handler.go

```diff
@@ -256,10 +257,18 @@ func (h *Handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 		span.AddAttributes(opencensusclient.EventTraceAttributes(event)...)
 	}
 
+	var scheme string
+	parsedUrl, err := url.Parse(event.Source())
+	if err != nil {
+		scheme = "unknown"
+	}
+	scheme = parsedUrl.Scheme
+
```

Comment:
Here, you should use the scheme from the request, not the event source. The event source is often not a simple url of the form `http://....` (see examples here: https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#source-1). It is simply a URI pointing to the source. For example, I could create an event with the source set to http://my-service, but then deliver the event using https to the broker. In this scenario the correct scheme would still be https, even though the source field is http. Similarly, the source could just be `/my-service/a`, in which case parsing the scheme would get us "unknown" here when I still could have delivered the event with http or https.

We should use the scheme of the request to know if this particular event (when bound to http) was delivered with http or https (or something else I guess)